lms/djangoapps/labster_course_license/licensed_blocks_override.py

```diff
@@ -0,0 +1,77 @@
+"""
+API related to providing field overrides for individual students.  This is used
+by the individual custom courses feature.
+"""
+import logging
+from courseware.field_overrides import FieldOverrideProvider
+from labster_course_license.models import LicensedCoursewareItems, CourseLicense
+from labster_course_license.utils import get_block_course_key
+from ccx_keys.locator import CCXLocator
+from ccx.overrides import get_current_ccx
+
+
+log = logging.getLogger(__name__)
+
+
+class LicensedBlocksOverrideProvider(FieldOverrideProvider):
+    """
+    A concrete implementation of
+    :class:`~courseware.field_overrides.FieldOverrideProvider` which allows for
+    overrides to be made on a per user basis.
+    """
+    def get(self, block, name, default):
+        """
+        Overrides `is_visible_to_staff_only` field of block depending on if licensed simulations in it.
+
+        Teacher or course staff can hide blocks in studio manually so we need to return `default` instead of `False`,
+        this will allow to stay in sync with studio visibility edits.
+        """
+        if name != 'visible_to_staff_only':
+            return default
+        course_key = get_block_course_key(block)
+        if course_key is not None:
+            # find the ccx that is active for block course
+            try:
+                ccx = get_current_ccx(course_key)
+                if ccx:
+                    ccx_key = CCXLocator.from_course_locator(course_key, ccx.id)
```

Comment:
`ccx = get_current_ccx(course_key)` raises exception
ccx_key is always not None